lib/scarlet/helpers/nick_helper.rb

```diff
@@ -0,0 +1,57 @@
+class Scarlet
+  module NickHelper
+    def handle_special_nick(name)
+      case name
+      when ':me'
+        sender.nick
+      when ':you'
+        server.current_nick
+      else
+        name
+      end
+    end
+
+    def find_nicks(names)
+      names.words.map do |name|
+        name = handle_special_nick(name)
+        Scarlet::Nick.first(nick: name)
+      end
+    end
+
+    def find_nick(name)
+      find_nicks(name).first
+    end
+
+    # Expects the command to have a :nick param
+    #
+    # @param [String] nn  nickname to look for, if nil, will expect a params[:nick]
+    # @yieldparam [Nick] nick
+    def with_nick(nn = nil, &block)
+      options = {
+        msgfmt: 'Cannot find Nick %s'
+      }
+      if nn.is_a?(Hash)
+        options = nn
```

Comment:
`options, nn = nn, options[:nick] if nn.is_a? Hash`

Not sure if it actually works :P